File: frontend/src/api/feedback.ts
/**
 * 反馈 API
 */
import { apiClient } from '@/utils/request'

export interface Feedback {
  id: number
  user_id: number
  username?: string
  title: string
  description: string
  images?: string[]
  status: 'pending' | 'processing' | 'resolved' | 'closed'
  priority: 'low' | 'medium' | 'high' | 'urgent'
  admin_reply?: string
  created_at: string
  updated_at: string
}

export interface FeedbackCreateParams {
  title: string
  description: string
  images?: string[]
}

export interface FeedbackUpdateParams {
  status?: string
  priority?: string
  admin_reply?: string
}

export interface FeedbackListParams {
  page?: number
  page_size?: number
  status?: string
  priority?: string
  user_id?: number
}

export interface FeedbackListResponse {
  items: Feedback[]
  total: number
  page: number
  page_size: number
}

export interface FeedbackStatistics {
  total: number
  by_status: Record<string, number>
  by_priority: Record<string, number>
}

/**
 * 上传反馈图片
 */
export const uploadImage = (file: File) => {
  const formData = new FormData()
  formData.append('file', file)
  return apiClient.post<{ url: string }>('/api/v1/feedback/upload-image', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  })
}

/**
 * 创建反馈
 */
export const createFeedback = (data: FeedbackCreateParams) => {
  return apiClient.post<Feedback>('/api/v1/feedback/create', data)
}

/**
 * 获取反馈列表
 */
export const getFeedbackList = (params?: FeedbackListParams) => {
  return apiClient.get<FeedbackListResponse>('/api/v1/feedback/list', { params })
}

/**
 * 获取反馈详情
 */
export const getFeedbackDetail = (id: number) => {
  return apiClient.get<Feedback>(`/api/v1/feedback/${id}`)
}

/**
 * 更新反馈（管理员）
 */
export const updateFeedback = (id: number, data: FeedbackUpdateParams) => {
  return apiClient.put<Feedback>(`/api/v1/feedback/${id}`, data)
}

/**
 * 删除反馈（管理员）
 */
export const deleteFeedback = (id: number) => {
  return apiClient.delete(`/api/v1/feedback/${id}`)
}

/**
 * 获取反馈统计（管理员）
 */
export const getFeedbackStatistics = () => {
  return apiClient.get<FeedbackStatistics>('/api/v1/feedback/statistics/summary')
}

File: backend/app/schemas/feedback.py
"""
用户反馈相关的 Schema
"""
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


class FeedbackCreateRequest(BaseModel):
    """创建反馈请求"""
    title: str = Field(..., min_length=1, max_length=200, description="反馈标题")
    description: str = Field(..., min_length=1, description="反馈描述")
    images: Optional[List[str]] = Field(default=None, description="图片URL列表")


class FeedbackUpdateRequest(BaseModel):
    """更新反馈请求（管理员）"""
    status: Optional[str] = Field(None, description="状态")
    priority: Optional[str] = Field(None, description="优先级")
    admin_reply: Optional[str] = Field(None, description="管理员回复")


class FeedbackResponse(BaseModel):
    """反馈响应"""
    id: int
    user_id: int
    username: Optional[str] = None  # 用户名（关联查询）
    title: str
    description: str
    images: Optional[List[str]] = None
    status: str
    priority: str
    admin_reply: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class FeedbackListResponse(BaseModel):
    """反馈列表响应"""
    items: List[FeedbackResponse]
    total: int
    page: int
    page_size: int
